klog/app/bookmark.go

```diff
@@ -213,3 +214,7 @@ func (bc *bookmarksCollection) ToJson() string {
 	}
 	return buffer.String()
 }
+
+func getDefaultBookmarkName() Name {
+	return NewName(os.Getenv("KLOG_DEFAULT_BOOKMARK"))
```

Comment:
Currently, the bookmark implementation inside `bookmark.do` is pure, i.e. it doesn’t perform side-effects such as an access to the program’s runtime environment.

The way klog’s code is designed right now, the proper place for an `os.GetEnv` call would be the very top-level [`klog.go` file](https://github.com/jotaen/klog/blob/592ebb9d64295981c8e6ac47a2345f4a6da850c0/klog.go#L43-L53). From there on, all environment-specific information is supposed to be passed down, e.g. to the [app context](https://github.com/jotaen/klog/blob/main/klog/app/context.go).

That being said, the required changes for this feature request would introduce a fair bit of complexity, unfortunately.

Reply:
Yeah, that makes sense. Would you mind taking a look at a second attempt where I try to make the necessary changes? If you still feel it adds unnecessary complexity after that, I'm fine in leaving this alone for now.

Reply:
I quickly threw together an implementation and it does the code a bit more involved since I had to extend the config with a `DefaultBookmark` option. I think the idea is correct, but I'll just honor your opinion here and let this go for now.